doc: Document OpenAPI generator + usage (#227)
SAP/cloud-sdk: docs/js/features/openapi/execute-openapi-request.mdx

---

import useBaseUrl from '@docusaurus/useBaseUrl';
import {
  CustomRequestConfig,
  CustomRequestConfigNote,
  ExecuteRawOverview,
  ExecuteRawUseCases
} from '../common';

:::note
The OpenAPI generator is currently beta.

> We do not recommend to use it in production as there might still be breaking changes.

Reviewer: I find this a bit hard.

Author: I am happy to adjust it, but as I don't find it that hard myself, I don't know how. Do you have a suggestion?

Reviewer 2: question: for the time being, this is still beta, but we will release a GA ~next week? This line will be removed then, right?

Query parameters are provided as an object, e.g. if you can set a `limit` parameter, `MyApi.getMyResource(resourceId, { limit: 10 })`.
The types of the parameters depend on the specification.
Their names are as specified in the original OpenAPI document.

#### Request Bodies

Request bodies can be both mandatory and optional.
The according function argument is always `body` and it is always between the path and query parameters, e.g. when the body is a simple object containing a title, `MyApi.updateMyResource(resourceId, { title: 'My Title' }, { lang: 'en' })`.
It the request body is optional, you have to pass `undefined`, e.g. `MyApi.updateMyResource(resourceId, undefined, { lang: 'en' })`.

### Setting Custom Headers

The SAP Cloud SDK sets all mandatory headers by default.
To set additional headers or change the default values used in the headers, use the `addCustomHeaders` method.
You can pass an object, where the key is the name of the header and the value, well the value.

```ts
MyApi.myFunction()
  .addCustomHeaders({
    apikey: 'my-api-key'
  })
  .execute(destination);
```

Custom headers take precedence over default headers.

> The example above can come in handy, when executing requests against the sandbox systems of the SAP SPI Business Hub.

Reviewer: Bit too casual.

> The example above can come in handy, when executing requests against the sandbox systems of the SAP SPI Business Hub.

Reviewer: ```suggestion
The example above can be used to execute requests against the sandbox systems of the SAP SPI Business Hub.
```
